test/js/specs/unit/collections/AccessPolicy.spec.js

```diff
@@ -0,0 +1,1519 @@
+        );
+        expect(allowsWithPerms.length).to.equal(1);
+
+        const allowNode = allowsWithPerms[0];
+        expect(allowNode.tagName.toLowerCase()).to.equal("allow");
+        expect(allowNode.querySelector("subject").textContent).to.equal("public");
+      });
+
+      it("maintains all access rules through serialize/parse round-trip", function() {
+        const policy = new AccessPolicy();
+        const originalRules = [
+          { subject: "uid=user1,dc=example", read: true, write: false, changePermission: true },
+          { subject: "public", read: true },
+          { subject: "CN=Group Name,DC=dataone,DC=org", read: true, write: true }
+        ];
+
+        policy.add(originalRules);
+
+        // Serialize to XML
+        const xml = policy.serialize();
+        expect(xml).to.not.be.null;
+
+        // Create new policy and parse the XML
+        const newPolicy = new AccessPolicy();
+        newPolicy.parse(xml);
+
+        // Should have same rules
+        expect(newPolicy.length).to.equal(originalRules.length);
+
+        // Verify each rule is preserved
+        originalRules.forEach(rule => {
+          const found = newPolicy.findWhere({ subject: rule.subject });
+          expect(found).to.exist;
+          expect(found.get("read")).to.equal(rule.read);
+        });
+      });
+    });
+
+    //-----------------------------------------------------------------------
+    // 3.9  getSubjectInfo()
+    //-----------------------------------------------------------------------
+    describe("getSubjectInfo()", function() {
+      it("forwards the call to each AccessRule instance", function() {
+        const ar1 = makeRule({ subject: "alice" });
+        const ar2 = makeRule({ subject: "bob" });
+        state.policy.add(ar1);
+        state.policy.add(ar2);
+
+        // Ensure the method exists, then spy on it
+        ar1.getSubjectInfo = ar1.getSubjectInfo || function() {
+        };
+        ar2.getSubjectInfo = ar2.getSubjectInfo || function() {
+        };
+        const spy1 = sinon.spy(ar1, "getSubjectInfo");
+        const spy2 = sinon.spy(ar2, "getSubjectInfo");
+
+        // Exercise
+        state.policy.getSubjectInfo();
+
+        // Verify
+        expect(spy1.called).to.be.true;
+        expect(spy2.called).to.be.true;
+
+        // Cleanup
+        spy1.restore();
+        spy2.restore();
+      });
+    });
+
+    //-----------------------------------------------------------------------
+    // 3.10  Event handling – “removeMe” propagation
+    //-----------------------------------------------------------------------
+    describe("event handling", function() {
+      it.skip("EXPECTED FAILURE: removes a rule when the rule fires a 'removeMe' event", function() {
+        const rule = makeRule({ subject: "temp" });
+        state.policy.add(rule);
+        expect(state.policy.length).to.equal(1);
+        rule.trigger("removeMe");
+        expect(state.policy.length).to.equal(0);
```

Comment:
```suggestion
        state.policy.listenToOnce(state.policy, "remove", () => {
          expect(state.policy.length).to.equal(0);
        });
        rule.trigger("removeMe");
```

The original test fails because it checks the collection length immediately after triggering "removeMe", before the **asynchronous** removal actually completes. Waiting for the "remove" event makes sure the assertion runs when the model has actually been removed. It was backbone's solution of ensuring async events had completed, back before we had Promises, await/async!